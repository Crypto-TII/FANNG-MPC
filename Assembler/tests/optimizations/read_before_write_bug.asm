# test_comparison-0--0
ldi c0, 2 # 0
ldsi s0, 1 # 1
ldsi s1, 1 # 2
movs s5, s0 # 3
movs s6, s1 # 4
modci c1, c0, 2 # 5
subc c4, c0, c1 # 6
ldi c13, 1 # 7
divci c14, c13, 2 # 8
mulc c12, c4, c14 # 9
modci c2, c12, 2 # 10
vmulci 2, c7, c1, 2 # 11
vmulm 2, s11, s5, c7 # 12
vaddm 2, s13, s5, c1 # 13
vsubs 2, s7, s13, s11 # 14
startopen 1, s14 # 15
stopopen 1, c37 # 16
stmc c37, 1005 # 17
vstartopen 2, 1, s7 # 18
vstopopen 2, 1, c38 # 19
stmc c38, 1006 # 20
reqbl 19 # 21
reqbl 19 # 22
