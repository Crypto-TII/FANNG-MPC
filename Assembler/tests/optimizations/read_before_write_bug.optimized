ldi c2, 2 # 0
ldsi s0, 1 # 1
ldsi s1, 1 # 2
ldi c3, 1 # 7
movs s2, s0 # 3
movs s3, s1 # 4
modci c4, c2, 2 # 5
divci c3, c3, 2 # 8
subc c2, c2, c4 # 6
mulc c2, c2, c3 # 9
modci c5, c2, 2 # 10
vaddm 2, s0, s2, c4 # 13
startopen 1, s1 # 15
vmulci 2, c6, c4, 2 # 11
vmulm 2, s2, s2, c6 # 12
vsubs 2, s0, s0, s2 # 14
stopopen 1, c2 # 16
stmc c2, 1005 # 17
reqbl 19 # 21
reqbl 19 # 22
vstartopen 2, 1, s0 # 18
vstopopen 2, 1, c0 # 19
stmc c0, 1006 # 20
