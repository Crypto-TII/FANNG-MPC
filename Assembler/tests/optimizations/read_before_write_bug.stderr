warning: Write to register is never read
  --> tests/optimizations/read_before_write_bug.asm:21:1
   |
21 | vstopopen 2, 1, c38 # 19
   | ------------------------ register 39 written to but will never be read from again
   |